src/extractor.rs

```diff
@@ -157,6 +162,35 @@ impl Arbitrary for ConfigVal {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum RefVersion {
+    StringRV(String),
+    IntRV(i32),
+    DoubleRV(f64),
+}
+
+#[cfg(test)]
+impl Arbitrary for RefVersion {
+    fn arbitrary(g: &mut Gen) -> RefVersion {
+        let kind: usize = usize::arbitrary(g) % 4;
+
+        match kind {
+            0 => RefVersion::StringRV(String::arbitrary(g)),
+            1 => RefVersion::IntRV(i32::arbitrary(g)),
+            2 => RefVersion::DoubleRV(f64::arbitrary(g)),
+            _ => panic!(),
+        }
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = RefVersion>> {
```

Comment:
`shrink` looks like it's a base case. If we try to shrink the "smallest" value it should return an empty iterator instead.